src/core/service.c

```diff
@@ -114,6 +115,8 @@ static void service_init(Unit *u) {
         s->timeout_abort_usec = u->manager->default_timeout_abort_usec;
         s->timeout_abort_set = u->manager->default_timeout_abort_set;
         s->restart_usec = u->manager->default_restart_usec;
+        s->restart_steps = 0;
```

Comment:
this is redundant. all fields are zero-initialized by default. we only need to initialize the fields that shall not be zero-initialized here.